### PR TITLE
[#31] Handling missing items

### PR DESCRIPTION
**Pull request by sarahwatremet**
# 🚀 Description

It works 🙌

## 👩‍🔬 Comments

Missing items are being displayed correctly.
No more loading issues.
User is redirected to home page when logged in.
Alert is displayed when user tries to update their collection while not logged in.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## 🗒 Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: AnimalPocket/AnimalPocket/ViewModels/UserViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+  
+  @Published var showMissingBugs = false
+  @Published var showMissingFish = false
+  @Published var showMissingSeaCreatures = false
+  
+  @Published var filter: Filter = .noFilter
+  
+  @Published var selectedTab = 4
+  
+  //  @MainActor
```
Comment (laure-sterne) [inline AnimalPocket/AnimalPocket/ViewModels/UserViewModel.swift:28]: C'est normal que ce soit commenté ? Est-ce qu'on l'intègre, on le supprime ou on le garde ?

Comment (sarahwatremet) [reply AnimalPocket/AnimalPocket/ViewModels/UserViewModel.swift:28]: J'ai hésité à virer le commentaire en effet, je l'ai gardé en me disant qu'on s'en servirait plus tard mais c'est pas ouf de garder un commentaire ! Qu'est-ce que tu en penses ?

Comment (laure-sterne) [reply AnimalPocket/AnimalPocket/ViewModels/UserViewModel.swift:28]: Si tu penses qu'on va sûrement l'utiliser plus tard, j'me dis qu'on peut le garder dans ce cas !

**Review by laure-sterne — APPROVED** [review]
That's wonderful! Great job!

**Pull request merged**